src/snowflake/connector/pandas_tools.py

```diff
@@ -142,6 +142,7 @@ def write_pandas(
     if chunk_size is None:
         chunk_size = len(df)
     cursor = conn.cursor()
+    cursor.execute(f'USE SCHEMA "{schema}"')
```

Comment:
@psukys This solution will, unfortunately, fail in case `schema` is not provided (see line 50: it is optional, so is `None` by default). A check `if schema:` prior to line 145 would fix this.

At the same time, the execution of "USE SCHEMA" will change the `schema` attribute in the `conn` object passed to the `write_pandas()` which might be considered an unwanted effect. Taking this into account, doesn't the proposed solution need a more global reconsideration and improvement?